public/docs/ts/latest/tutorial/toh-pt5.jade

```diff
@@ -530,14 +530,27 @@ block ngOnInit
 
 block extract-id
   :marked
-    Notice how we extract the `id` by calling the `forEach` method
-    which will deliver our !{_array} of route parameters.
+    Notice how we use the `switchMap` operator on the `Observable` to
```

Comment:
I wonder why the original said there is an array of route parameters. There is a params _object_ and the  `+params['id']` expression gets the `id` parameter from that object. Weird.

Rewrite
...
Note how the `switchMap` operator maps the `id` in the observable route parameters
to a new `Observable`, the result of the  `HeroService.getHero` method.

If the user re-navigates to this component while a `getHero` request is still inflight,
`switchMap` cancels that old request before calling  `HeroService.getHero` again.
...